src/v2/Components/ArtworkFilter/ArtworkFilters/__tests__/SizeFilter2.rtl.jest.tsx

```diff
@@ -0,0 +1,171 @@
+
+const Wrapper: React.FC = ({ children }) => {
+  return (
+    <ArtworkFilterContextProvider>
+      <ClearAllButton />
+      {children}
+      <ArtworkFilterContextInspector />
+    </ArtworkFilterContextProvider>
+  )
+}
+
+const ClearAllButton: React.FC = () => {
+  const artworkFilterContext = useArtworkFilterContext()
+
+  return (
+    <button onClick={() => artworkFilterContext.resetFilters()}>
+      Clear all
+    </button>
+  )
+}
+
+const ArtworkFilterContextInspector: React.FC = () => {
+  const artworkFilterContext = useArtworkFilterContext()
+
+  return (
+    <aside data-testid="artwork-filter-context-inspector">
+      {JSON.stringify(artworkFilterContext, null, 2)}
+    </aside>
+  )
+}
+
+const currentContext = (): ArtworkFilterContextProps => {
+  let contextInspector: HTMLElement
+  try {
+    contextInspector = screen.getByTestId("artwork-filter-context-inspector")
+  } catch (error) {
+    if (error.name === "TestingLibraryElementError")
+      throw new Error(
+        `The currentContext() helper function requires an <ArtworkFilterContextInspector /> to be mounted in the current DOM.`
+      )
+  }
+  return JSON.parse(contextInspector!.textContent!)
+}
+
+describe("SizeFilter2", () => {
+  it("toggles custom input render", () => {
+    render(<SizeFilter2 expanded />)
+
+    // note: an <input type="number" /> has a role of 'spinbutton'
+    expect(screen.queryByRole("spinbutton")).not.toBeInTheDocument()
+
+    userEvent.click(screen.getByText("Show custom size"))
+
+    expect(screen.getByText("Hide custom size")).toBeInTheDocument()
+    expect(screen.getAllByRole("spinbutton")).toHaveLength(4)
+  })
+
+  it("updates context on filter change", () => {
+    render(<SizeFilter2 expanded />)
+    expect(currentContext().filters?.sizes).toEqual([])
+
+    userEvent.click(screen.getAllByRole("checkbox")[0])
+    expect(currentContext().filters?.sizes).toEqual(["SMALL"])
+
+    userEvent.click(screen.getAllByRole("checkbox")[2])
+    expect(currentContext().filters?.sizes).toEqual(["SMALL", "LARGE"])
+  })
+
+  it("updates the filter values", async () => {
+    render(<SizeFilter2 expanded />)
+
+    userEvent.click(screen.getByText("Show custom size"))
+    userEvent.type(screen.getAllByRole("spinbutton")[0], "12")
+    userEvent.type(screen.getAllByRole("spinbutton")[1], "16")
+    userEvent.type(screen.getAllByRole("spinbutton")[2], "12")
+    userEvent.type(screen.getAllByRole("spinbutton")[3], "16")
+    userEvent.click(screen.getByText("Set size"))
+
+    expect(currentContext().filters?.sizes).toEqual([])
+    expect(currentContext().filters?.height).toEqual("4.72-6.3")
+    expect(currentContext().filters?.width).toEqual("4.72-6.3")
+  })
+
+  it("updates the filter values when only one dimension is added", () => {
+    render(<SizeFilter2 expanded />)
+    userEvent.click(screen.getByText("Show custom size"))
+    screen.getAllByRole("spinbutton").map(field => userEvent.clear(field)) // 😭
```

Comment:
Here and L127 you can see the one goofy workaround I needed in order to get around the problem of state leaking from one test to another — somehow the input fields retain the values that are "typed" in from earlier specs, so I manually cleared them 😕